Implementation plan  for security baseline SIG onboarding

Danajoyluck opened the pull request:
Detailed implementation plan for onboarding security baseline SIG.

Danajoyluck committed:
Implementation plan  for security baseline SIG onboarding 

Detailed implementation plan for onboarding security baseline SIG. 

Signed-off-by: Dana Wang <Danajoyluck@users.noreply.github.com>

david-a-wheeler commented on operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:
```diff
@@ -0,0 +1,45 @@
+* Validate the above artifacts are created successfully
+
+## Prerequisites
+* [Prerequist](Prerequisites) in the onboadring radme.md
```
Spelling nit: "onboadring radme.md" should be "onboarding readme.md"

Danajoyluck replied:
checked in update. Thank you @david-a-wheeler.

david-a-wheeler commented on operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:
```diff
@@ -0,0 +1,45 @@
+2. Run below command to
+* Create teams, 1 main team  and 2 sub teams with different repo roles
+* Add members to different teams, assign them team roles
+* Creat repo
```
Nit: "Creat" -> "Create"

Danajoyluck replied:
checked in update. Thank you @david-a-wheeler.

david-a-wheeler commented on operations/github/onboarding/implementation_plan/security_baseline_sig_onboarding.md:
```diff
@@ -0,0 +1,45 @@
+* Creat repo
+* Assign teams' access to the repo
+* Create a test issue
+* Look up all the artigfacts created from above and write the search results to security_baseline_results.json
```
artigfacts ?

Danajoyluck replied:
checked in update. Thank you @david-a-wheeler.

david-a-wheeler commented:
While reading I noted some spelling nits (see above), but really, this looks fine.

This is for simply onboarding / creating the infrastructure for the SIG (mainly its repo) itself, and that's fine. Its README will need to make it clear who the leaders are, and that it's part of the Best Practices WG. The WG should also point to this, once it's set up.

An alternative would be to house this *within* the best practices WG. The WG already has a setup to create pretty web pages, while if you want to do that in this SIG, there may be an extra step. However, because there are a number of other foundations involved, I think the approach (as proposed) is the better way anyway.

We may eventually want to create materials that will *implement* the baseline. That work might be part of this repo, or a separate repo. I would suggest putting it all in this repo if that makes sense, and if we discover it doesn't, we'll have a rationale for creating another repo.

Thanks for doing this!

Danajoyluck committed:
Update security_baseline_sig_onboarding.md to fix typos

Fixed the typos that @david-a-wheeler pointed out.

Signed-off-by: Dana Wang <Danajoyluck@users.noreply.github.com>

Commit:
Updated the code to fix a bug that caused the HTTP response code not properly returned. Also cleaned up the code to make terminal output cleaner

Signed-off-by: Dana Wang <dwang@linuxfoundation.org>

Danajoyluck committed:
Update security_baseline_sig_onboarding.md

Updated the link to prerequisite that references the onboarding readme.md Prerequisites

Updated the main team and maintainer team to have @david-a-wheeler as the maintainer instead of a member. 

Signed-off-by: Dana Wang <Danajoyluck@users.noreply.github.com>

Commit:
Updated the main team parent to be SIGs per TAC policy, updated David and myself to be a maintainer of each team

Signed-off-by: Dana Wang <dwang@linuxfoundation.org>

bbpursell1 commented on operations/github/onboarding/repo_init_metadata/security_baseline.json:
```diff
@@ -8,7 +8,7 @@
+      "parent_team": "SIGs",
       "privacy": "closed",
       "notification_setting": "true",
       "permission": "pull",
```
Is this a valid permission?  I looks like you're using this in the call to this https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#add-or-update-team-project-permissions

That lists only "read, write, admin" as the possible values.  I think the "pull" value is deprecated.  I would assume that this should allow for the "maintain" role, but it isn't in the documentation.

Danajoyluck replied:
this is the API that the code uses to add or update repo access
https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#add-or-update-team-repository-permissions

bbpursell1 reviewed (approved):
After discussion and clarification on the permissions, this looks good.
